File: Reza/Codeforces/python/x703A.py
n = int(input())
m_c = 0
c_c = 0

for _ in range(n):
    dices = list(map(int, input().split(" ")))
    if dices[0] > dices[1]:
        m_c += 1
    elif dices[0] < dices[1]:
        c_c += 1

if m_c == c_c:
    print("Friendship is magic!^^")
elif m_c > c_c:
    print("Mishka")
else:
    print("Chris")

File: Ali/Java/X703A.java

import java.util.Scanner;

public class X703A {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int mishka = 0;
        int chris = 0;

        while (n-- > 0) {
            int mi = input.nextInt();
            int ci = input.nextInt();

            if (mi > ci) {
                mishka++;
            } else if (mi < ci) {
                chris++;
            }
        }
        if (mishka > chris) {
            System.out.println("Mishka");
        } else if (mishka < chris) {
            System.out.println("Chris");
        } else {
            System.out.println("Friendship is magic!^^");
        }
    }
}
